src/main/resources/api.yaml

```diff
@@ -30,6 +29,66 @@ paths:
             $ref: '#/definitions/LoginResult'
         400:
           description: Validation error
+  /projects:
+    get:
+      tags:
+        - projects
+      summary: Retrieves a list of the current user's projects
+      operationId: getProjects
+      produces:
+        - application/json
+      responses:
+        200:
+          description: Projects retrieved successfully
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/ProjectDto'
+    post:
+      tags:
+        - projects
+      summary: Creates a new project
+      operationId: addProject
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - in: body
+          name: request
+          description: The project
+          required: true
+          schema:
+            $ref: '#/definitions/ProjectDto'
```

Comment:
A jeszcze taka myśl - to zakłada PARSOWANIE DOUBLE'A po stronie frontu?

Reply:
Nawet nie jestem pewny czy wymaga parsowania double'a, chyba jak dasz `"1.23"` zamiast `1.23` to też przejdzie :smiley:  Ale w każdym razie zakłada walidację, bo walidacja musi być, także niech front się strzeże.

Reply:
Ale jak napiszesz "jeden kropka dwadzieścia trzy", to może być problem

Reply:
Kiedyś coś chyba mówiliśmy, że FRONT MA NIC NIE PARSOWAĆ/WALIDOWAĆ

Reply:
Ale nie pamiętam kto, jak, gdzie i dlaczego

Reply:
Więc chyba się nie liczy

Reply:
Raczej podstawowe walidacje w stylu "czy to liczba" lub "czy nie jest puste" to się robi na froncie, dzięki temu user ma szybki feedback i API nie jest skomplikowane.

Reply:
Spoks, ja nie mam z tym problemu, tylko pamiętam, że był taki postulat

Reply:
W kontrolkach do wpisywania można ustalić jaki jest format danych, np. regexem
Nie uwzględnia to jednak podania złych danych przez jakieś inne żądanie (niekoniecznie przez aplikację)

Wydaje mi się, że jak się zamiast inta wyśle "jeden kropka dwadzieścia trzy" to poleci wyjątek, bo nie uda się przypisanie tego do zmiennej w metodzie (ale trzeba by sprawdzić, czy tak faktycznie jest)

Reply:
@szbrnl jak się da `"0.7"` zamiast `0.7` to działa, właśnie potwierdziłem, jak się da złą wartość numeryczną w stringu to automatycznie 400 leci